lib/sql-string.js

```diff
@@ -35,6 +35,11 @@ function escape(val, timeZone, dialect, format) {
     break;
   }
 
+  // moment
+  if (val.toDate) {
```

Comment:
I generally disagree with @sushantdhiman, duck typing should always be preferred. But in this case the code should just be refactored so that the input is directly passed to `moment`, which accepts a `Date`, a string or a `moment`.